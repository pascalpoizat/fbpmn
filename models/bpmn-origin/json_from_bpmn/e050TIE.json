{
    "isInterrupt": {},
    "catN": {
        "StartEvent_1": "NoneStartEvent",
        "Process_1": "Process",
        "EndEvent_14srvev": "NoneEndEvent",
        "IntermediateThrowEvent_1l1btga": "TimerIntermediateEvent"
    },
    "catE": {
        "SequenceFlow_04ghp5z": "NormalSequenceFlow",
        "SequenceFlow_0yeopwj": "NormalSequenceFlow"
    },
    "containN": {
        "Process_1": [
            "EndEvent_14srvev",
            "StartEvent_1",
            "IntermediateThrowEvent_1l1btga"
        ]
    },
    "messageE": {},
    "timeInformation": {
        "IntermediateThrowEvent_1l1btga": {
            "timerDefinitionValue": null,
            "timerDefinitionType": null
        }
    },
    "targetE": {
        "SequenceFlow_04ghp5z": "EndEvent_14srvev",
        "SequenceFlow_0yeopwj": "IntermediateThrowEvent_1l1btga"
    },
    "name": "e050TIE",
    "attached": {},
    "containE": {
        "Process_1": [
            "SequenceFlow_0yeopwj",
            "SequenceFlow_04ghp5z"
        ]
    },
    "edges": [
        "SequenceFlow_0yeopwj",
        "SequenceFlow_04ghp5z"
    ],
    "messages": [],
    "sourceE": {
        "SequenceFlow_04ghp5z": "IntermediateThrowEvent_1l1btga",
        "SequenceFlow_0yeopwj": "StartEvent_1"
    },
    "nodes": [
        "Process_1",
        "EndEvent_14srvev",
        "StartEvent_1",
        "IntermediateThrowEvent_1l1btga"
    ],
    "nameN": {}
}