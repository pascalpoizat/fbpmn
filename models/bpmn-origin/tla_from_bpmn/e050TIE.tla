---------------- MODULE e050TIE ----------------

EXTENDS TLC, PWSTypes

VARIABLES nodemarks, edgemarks, net

Interest ==
  "Process_1" :> {  }

ContainRel ==
  "Process_1" :> { "EndEvent_14srvev", "StartEvent_1", "IntermediateThrowEvent_1l1btga" }

Node == { "Process_1", "EndEvent_14srvev", "StartEvent_1", "IntermediateThrowEvent_1l1btga" }

Edge == { "SequenceFlow_0yeopwj", "SequenceFlow_04ghp5z" }

Message == {  }

msgtype ==
  [ i \in {} |-> {}]


source ==
   "SequenceFlow_0yeopwj" :> "StartEvent_1"
@@ "SequenceFlow_04ghp5z" :> "IntermediateThrowEvent_1l1btga"

target ==
   "SequenceFlow_0yeopwj" :> "IntermediateThrowEvent_1l1btga"
@@ "SequenceFlow_04ghp5z" :> "EndEvent_14srvev"

CatN ==
   "Process_1" :> Process
@@ "EndEvent_14srvev" :> NoneEndEvent
@@ "StartEvent_1" :> NoneStartEvent
@@ "IntermediateThrowEvent_1l1btga" :> TimerIntermediateEvent

CatE ==
   "SequenceFlow_0yeopwj" :> NormalSeqFlow
@@ "SequenceFlow_04ghp5z" :> NormalSeqFlow

LOCAL preEdges ==
  [ i \in {} |-> {}]
PreEdges(n,e) == preEdges[n,e]

PreNodes(n,e) == { target[ee] : ee \in preEdges[n,e] }
          \union { nn \in { source[ee] : ee \in preEdges[n,e] } : CatN[nn] \in { NoneStartEvent, MessageStartEvent } }

BoundaryEvent ==
  [ i \in {} |-> {}]

WF == INSTANCE PWSWellFormed
ASSUME WF!WellTyped
ASSUME WF!WellFormedness

ConstraintNode == TRUE \* none
ConstraintEdge == TRUE \* none
Constraint == TRUE     \* none
INSTANCE PWSConstraints
INSTANCE UserProperties
INSTANCE PWSSemantics

================================================================

