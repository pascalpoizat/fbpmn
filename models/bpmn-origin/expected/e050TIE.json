{
  "e050TIE": {
    "Network02FifoPair": {
      "Prop02Safe": {
        "value": true,
        "states": "4",
        "trans": "4",
        "depth": "4"
      },
      "Prop04MsgSound": {
        "value": true,
        "states": "4",
        "trans": "4",
        "depth": "4"
      },
      "Prop03Sound": {
        "value": true,
        "states": "4",
        "trans": "4",
        "depth": "4"
      },
      "Prop01Type": {
        "value": true,
        "states": "4",
        "trans": "4",
        "depth": "4"
      }
    },
    "Network03Causal": {
      "Prop02Safe": {
        "value": true,
        "states": "4",
        "trans": "4",
        "depth": "4"
      },
      "Prop04MsgSound": {
        "value": true,
        "states": "4",
        "trans": "4",
        "depth": "4"
      },
      "Prop03Sound": {
        "value": true,
        "states": "4",
        "trans": "4",
        "depth": "4"
      },
      "Prop01Type": {
        "value": true,
        "states": "4",
        "trans": "4",
        "depth": "4"
      }
    },
    "Network07RSC": {
      "Prop02Safe": {
        "value": true,
        "states": "4",
        "trans": "4",
        "depth": "4"
      },
      "Prop04MsgSound": {
        "value": true,
        "states": "4",
        "trans": "4",
        "depth": "4"
      },
      "Prop03Sound": {
        "value": true,
        "states": "4",
        "trans": "4",
        "depth": "4"
      },
      "Prop01Type": {
        "value": true,
        "states": "4",
        "trans": "4",
        "depth": "4"
      }
    },
    "Network05Outbox": {
      "Prop02Safe": {
        "value": true,
        "states": "4",
        "trans": "4",
        "depth": "4"
      },
      "Prop04MsgSound": {
        "value": true,
        "states": "4",
        "trans": "4",
        "depth": "4"
      },
      "Prop03Sound": {
        "value": true,
        "states": "4",
        "trans": "4",
        "depth": "4"
      },
      "Prop01Type": {
        "value": true,
        "states": "4",
        "trans": "4",
        "depth": "4"
      }
    },
    "Network04Inbox": {
      "Prop02Safe": {
        "value": true,
        "states": "4",
        "trans": "4",
        "depth": "4"
      },
      "Prop04MsgSound": {
        "value": true,
        "states": "4",
        "trans": "4",
        "depth": "4"
      },
      "Prop03Sound": {
        "value": true,
        "states": "4",
        "trans": "4",
        "depth": "4"
      },
      "Prop01Type": {
        "value": true,
        "states": "4",
        "trans": "4",
        "depth": "4"
      }
    },
    "Network01Bag": {
      "Prop02Safe": {
        "value": true,
        "states": "4",
        "trans": "4",
        "depth": "4"
      },
      "Prop04MsgSound": {
        "value": true,
        "states": "4",
        "trans": "4",
        "depth": "4"
      },
      "Prop03Sound": {
        "value": true,
        "states": "4",
        "trans": "4",
        "depth": "4"
      },
      "Prop01Type": {
        "value": true,
        "states": "4",
        "trans": "4",
        "depth": "4"
      }
    },
    "Network06Fifo": {
      "Prop02Safe": {
        "value": true,
        "states": "4",
        "trans": "4",
        "depth": "4"
      },
      "Prop04MsgSound": {
        "value": true,
        "states": "4",
        "trans": "4",
        "depth": "4"
      },
      "Prop03Sound": {
        "value": true,
        "states": "4",
        "trans": "4",
        "depth": "4"
      },
      "Prop01Type": {
        "value": true,
        "states": "4",
        "trans": "4",
        "depth": "4"
      }
    }
  }
}
